src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -358,10 +358,50 @@ type AdaptiveFSharpLspServer
             | Some false -> None
             | None -> None
 
+          // Stage 1 & 2: Use workspaceFolders instead of deprecated RootPath/RootUri
+          // Probe all workspace folders for F# code and select the first one with F# projects
           let actualRootPath =
-            match p.RootUri with
-            | Some rootUri -> Some(Path.FileUriToLocalPath rootUri)
-            | None -> p.RootPath
+            match p.WorkspaceFolders with
+            | Some(NonEmptyArray folders) ->
+              // Stage 2: Probe each workspace folder for F# code
+              let folderPaths =
+                folders |> Array.map (fun f -> Path.FileUriToLocalPath f.Uri) |> Array.toList
```

Comment:
Is there a need to convert the array into a list? (It looks like all it does is search the collection, which could use Array.tryFind rather than list.tryFind?)